modules/geo/src/internalClusterTest/java/org/opensearch/geo/search/aggregations/bucket/GeoTileGridIT.java

```diff
@@ -0,0 +1,249 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.geo.search.aggregations.bucket;
+
+import com.carrotsearch.hppc.ObjectIntHashMap;
+import com.carrotsearch.hppc.ObjectIntMap;
+import org.hamcrest.MatcherAssert;
+import org.opensearch.Version;
+import org.opensearch.action.index.IndexRequestBuilder;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.common.geo.GeoBoundingBox;
+import org.opensearch.common.geo.GeoPoint;
+import org.opensearch.common.geo.GeoShapeDocValue;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.geo.search.aggregations.bucket.geogrid.GeoGrid;
+import org.opensearch.geo.search.aggregations.bucket.geogrid.GeoGridAggregationBuilder;
+import org.opensearch.geo.search.aggregations.common.GeoBoundsHelper;
+import org.opensearch.geo.tests.common.AggregationBuilders;
+import org.opensearch.geo.tests.common.RandomGeoGenerator;
+import org.opensearch.geo.tests.common.RandomGeoGeometryGenerator;
+import org.opensearch.geometry.Geometry;
+import org.opensearch.geometry.Rectangle;
+import org.opensearch.search.aggregations.InternalAggregation;
+import org.opensearch.search.aggregations.bucket.GeoTileUtils;
+import org.opensearch.test.OpenSearchIntegTestCase;
+import org.opensearch.test.VersionUtils;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Random;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertAcked;
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertSearchResponse;
+
+@OpenSearchIntegTestCase.SuiteScopeTestCase
+public class GeoTileGridIT extends AbstractBucketAggregationIntegTest {
+
+    private static final int GEOPOINT_MAX_PRECISION = 17;
+
+    private static final String AGG_NAME = "geotilegrid";
+
+    private static final String GEO_SHAPE_INDEX_NAME = "geoshape_index";
+
+    private static final String GEO_SHAPE_FIELD_NAME = "location_geo_shape";
+
+    private final Version version = VersionUtils.randomIndexCompatibleVersion(random());
+
+    private static final Rectangle BOUNDING_RECTANGLE_FOR_GEO_SHAPES_AGG = new Rectangle(-4.1, 20.9, 21.9, -3.1);
```

Comment:
same as above.. abstracting a base GridIntegTest will allow us to reuse the logic, and extend for general XYShape in the future.

Reply:
Yes sure will do that.

Reply:
Actually this code was already abstracted it just I was not using it for this test case. Will fix it.

Reply:
The abstraction is already done via AbstractBucketAggregationIntegTest.class

Reply:
The abstraction is already done via AbstractBucketAggregationIntegTest.class